src/logger.js

```diff
@@ -1,23 +1,51 @@
 import createDebug from 'debug'
 import { discoveryKey } from 'hypercore-crypto'
+import mapObject from 'map-obj'
+import util from 'util'
 
 const TRIM = 7
 
-createDebug.formatters.h = (v) => {
+createDebug.formatters.h = function (v) {
   if (!Buffer.isBuffer(v)) return '[undefined]'
   return v.toString('hex').slice(0, TRIM)
 }
 
-createDebug.formatters.S = (v) => {
+createDebug.formatters.S = function (v) {
   if (typeof v !== 'string') return '[undefined]'
   return v.slice(0, 7)
 }
 
-createDebug.formatters.k = (v) => {
+createDebug.formatters.k = function (v) {
   if (!Buffer.isBuffer(v)) return '[undefined]'
   return discoveryKey(v).toString('hex').slice(0, TRIM)
 }
 
+/**
+ * @param {import('./sync/sync-state.js').State} v
+ * @this {any} */
+createDebug.formatters.X = function (v) {
```

Comment:
probably better to do as a follow-up but would be helpful to add a comment about what these formatters should be used for